### PR TITLE
Optimization of ScatterElementsUpdate reference implementation

### PR DESCRIPTION
**Pull request by tomdol**
### Details:
 - The rounding direction setting is moved out to a helper struct
 - This makes only 2 calls to the rounding mode related functions instead of 2 calls per each value
 - The code only takes effect for integral types and MEAN reduction mode

### CODE REVIEW

File: src/core/reference/include/ngraph/runtime/reference/scatter_elements_update.hpp [hunk 1]
```diff
@@ -165,6 +162,25 @@ size_t normalize_index(const T idx, const size_t dim_value) {
     }
 }
 
+template <typename T>
+struct RoundingDirectionGuard {
+    RoundingDirectionGuard() {
+        if (std::is_integral<T>::value) {
+            m_original_mode = std::fegetround();
+            std::fesetround(FE_DOWNWARD);
```
Comment (ilya-lavrenov) [inline src/core/reference/include/ngraph/runtime/reference/scatter_elements_update.hpp:170]: is it thread safe?

File: src/core/reference/include/ngraph/runtime/reference/scatter_elements_update.hpp [hunk 2]
```diff
@@ -221,6 +237,9 @@ void scatter_elem_update_with_reduction(const DataType* input_data,
     }
 
     if (reduction_type == Reduction::MEAN) {
+        // this object will change the rounding mode only for integer types which is required to match torch
```
Comment (rkazants) [inline src/core/reference/include/ngraph/runtime/reference/scatter_elements_update.hpp:240]: why does it mention torch? Will we be aligned with other FWs?

Comment (mitruska) [reply src/core/reference/include/ngraph/runtime/reference/scatter_elements_update.hpp:240]: Based on @tomdol https://github.com/openvinotoolkit/openvino/pull/18069#discussion_r1236760148 from the core op PR:
> The `mean` reduction type is only present in pytorch and it always rounds down the calculated value to the nearest integer. We've added this reduction type to OV to cover this case in PT FE and I think it should be implemented accordingly.

File: src/core/reference/include/ngraph/runtime/reference/scatter_elements_update.hpp (hunk 1 above)
Comment (mitruska) [reply src/core/reference/include/ngraph/runtime/reference/scatter_elements_update.hpp:170]: "It is thread-local. Each thread inherits the initial state of its floating-point environment from the parent thread."
Source: https://en.cppreference.com/w/cpp/numeric/fenv

**Review by mitruska — APPROVED** [review]

**Review by t-jankowski — APPROVED** [review]

**Pull request merged**